### ISSUE TITLE
email sending code may not be able to deal with unicode correctly

### ISSUE THREAD

**Issue by binary1230** [magfest/ubersystem#1958]
in testing, someone used unicode characters for their attendee name

you can see it here: UnicodeEncodeError: 'ascii' codec can't encode characters in position 263-268

```
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.4/logging/__init__.py", line 980, in emit
    stream.write(msg)
UnicodeEncodeError: 'ascii' codec can't encode characters in position 263-268: o
rdinal not in range(128)
Call stack:
  File "/usr/lib/python3.4/threading.py", line 888, in _bootstrap
    self._bootstrap_inner()
  File "/usr/lib/python3.4/threading.py", line 920, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.4/threading.py", line 868, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/local/uber/sideboard/lib/_threads.py", line 28, in run
    self.func()
  File "/usr/local/uber/plugins/uber/uber/automated_emails.py", line 72, in send
_all
    rem.send(inst, raise_errors=raise_errors)
  File "/usr/local/uber/plugins/uber/uber/automated_emails.py", line 54, in send
    send_email(self.sender, x.email, self.subject, self.render(x), format, model
=x, cc=self.cc, ident=self.ident)
  File "/usr/local/uber/plugins/uber/uber/utils.py", line 151, in send_email
Arguments: {'cc': [], 'to': [], 'ident': 'MAGFest payment received', 'source': '
MAGFest Registration <regsupport@magfest.org>', 'format': 'html', 'xs': [], 'mod
el': <Attendee full_name='\U0001f389\U0001f389\U0001f389\U0001f389\U0001f389\U00
01f389 \U0001f332\U0001f332\U0001f332\U0001f332\U0001f332\U0001f332\U0001f332'>,
 'dest': 'ttea@test.com', 'subject': 'MAGFest payment received', 'bcc': [], 'bod
y': b'<html>\n<head></head>\n<body>\n\nYou (\xf0\x9f\x8e\x89\xf0\x9f\x8e\x89\xf0
\x9f\x8e\x89\xf0\x9f\x8e\x89\xf0\x9f\x8e\x89\xf0\x9f\x8e\x89 \xf0\x9f\x8c\xb2\xf0\x9f\x8c\xb2\xf0\x9f\x8c\xb2\xf0\x9f\x8c\xb2\xf0\x9f\x8c\xb2\xf0\x9f\x8c\xb2\xf0\x9f\x8c\xb2) have preregistered for MAGFest this coming January 5-8\n\n
```

**Comment by binary1230** [comment]
not a reg launch blocker

**Comment by kitsuta** [comment]
I can no longer reproduce this bug and I think we actually fixed it when we moved from Django to Jinja2.

What I found is that this kind of bug is from using a `str` object to implicitly decode unicode ([source](https://stackoverflow.com/questions/9942594/unicodeencodeerror-ascii-codec-cant-encode-character-u-xa0-in-position-20)). In Django, the `SafeString` object -- what we used for the email body before we switched to Jinja2 -- is a subclass of `str` ([source](https://docs.djangoproject.com/en/1.11/ref/utils/#django.utils.safestring.SafeString)). However, when we converted to Jinja2, we made our own custom tag `safe_string` that casts the input into a `Markup` object if it isn't already one. And in Jinja2, the `Markup` object is a direct subclass of `unicode` ([source](http://jinja.pocoo.org/docs/2.9/api/#jinja2.Markup)). So now we happen to work entirely in unicode, and my tests on the staging server showed that I could use emoji or emoticons in the attendee name with no problem.
